src/parser.js

```diff
@@ -386,24 +437,114 @@ export default class Parser extends Stream {
+                  missingAttributes.push(k);
+                }
+              });
+
+              if (missingAttributes.length) {
+                const index = this.manifest.parts.length - 1;
+
+                this.trigger('warn', {
+                  message: `#EXT-X-PART #${index} lacks required attribute(s) ${missingAttributes.join(', ')}`
+                });
+              }
+
+              if (this.manifest.renditionReports) {
+                this.manifest.renditionReports.forEach((r, i) => {
+                  if (!r.hasOwnProperty('LAST-PART')) {
+                    this.trigger('warn', {
+                      message: `#EXT-X-RENDITION-REPORT #${i} lacks required attribute(s) LAST-PART`
+                    });
+                  }
+                });
+              }
             },
             'server-control'() {
-              this.manifest.serverControl = entry.attributes;
+              const attrs = entry.attributes;
+
+              this.manifest.serverControl = attrs;
+              if (!attrs.hasOwnProperty('CAN-BLOCK-RELOAD')) {
+                this.manifest.serverControl['CAN-BLOCK-RELOAD'] = false;
+                this.trigger('info', {
+                  message: '#EXT-X-SERVER-CONTROL defaulting CAN-BLOCK-RELOAD to false'
+                });
+              }
+              setHoldBack.call(this, this.manifest);
+
+              if (attrs.hasOwnProperty('CAN-SKIP-DATERANGES') && !attrs.hasOwnProperty('CAN-SKIP-UNTIL')) {
+                this.trigger('warn', {
+                  message: '#EXT-X-SERVER-CONTROL lacks required attribute CAN-SKIP-UNTIL which is required when CAN-SKIP-DATERANGES is set'
+                });
+
+              }
             },
             'preload-hint'() {
               this.manifest.preloadHints = this.manifest.preloadHints || [];
               this.manifest.preloadHints.push(entry.attributes);
+
+              const missingAttributes = [];
+
+              ['TYPE', 'URI'].forEach(function(k) {
+                if (!entry.attributes[k]) {
+                  missingAttributes.push(k);
+                }
+              });
+
+              if (missingAttributes.length) {
+                const index = this.manifest.preloadHints.length - 1;
+
+                this.trigger('warn', {
+                  message: `#EXT-X-PRELOAD-HINT #${index} lacks required attribute(s) ${missingAttributes.join(', ')}`
+                });
+              }
             },
             'rendition-report'() {
               this.manifest.renditionReports = this.manifest.renditionReports || [];
               this.manifest.renditionReports.push(entry.attributes);
+              const index = this.manifest.renditionReports.length - 1;
+              const missingAttributes = [];
+              const warning = `#EXT-X-RENDITION-REPORT #${index} lacks required attribute(s)`;
+
+              ['LAST-MSN', 'URI'].forEach(function(k) {
+                if (!entry.attributes[k]) {
```

Comment:
here as well, may be better to use `k in entry.attributes` as LAST-MSN is a number and could theoretically be zero.